### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by donije09**

### CODE REVIEW

File: .gitignore
Comment (truetechcode) [inline .gitignore:1]: Fix conflicts in this file

File: package-lock.json
Comment (truetechcode) [inline package-lock.json:1]: Delete this file

File: package.json
Comment (truetechcode) [inline package.json:1]: Fix conflicts in this file

File: src/components/main-view/main-view.jsx
Comment (truetechcode) [inline src/components/main-view/main-view.jsx:1]: Fix conflicts in this file

File: src/components/movie-card/movie-card.jsx
Comment (truetechcode) [inline src/components/movie-card/movie-card.jsx:1]: Fix conflicts in this file

File: src/components/movie-view/movie-view.jsx
Comment (truetechcode) [inline src/components/movie-view/movie-view.jsx:1]: Fix conflicts in this file

File: src/index.jsx
Comment (truetechcode) [inline src/index.jsx:1]: Fix conflicts in this file

**Review by truetechcode — CHANGES_REQUESTED** [review]
I have marked some files with merge conflicts that you must fix before you proceed, and you should also delete the `package-lock.json` file completely.

If you don't know how to fix merge conflicts then follow this guide on [Youtube](https://www.youtube.com/watch?v=lz5OuKzvadQ)

File: src/components/movie-view/movie-view.jsx [hunk 1]
```diff
@@ -1,26 +1,24 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-const MovieView = ({ movie, onBackClick }) => {
+import { Card, Button } from 'react-bootstrap';
+import { useParams, useNavigate } from 'react-router-dom';
```
Comment (truetechcode) [inline src/components/movie-view/movie-view.jsx:3]: You're using `react-router-dom` here as required, did you install it and how did you do it?

File: src/index.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { createRoot } from 'react-dom/client';
 import { MainView } from './components/main-view/main-view'; 
 import 'bootstrap/dist/css/bootstrap.min.css'; // Added import for Bootstrap styles
```
Comment (truetechcode) [inline src/index.jsx:4]: You have this line, but before this works you must first install `bootstrap`. Do that and let me know how you did it.

**Review by truetechcode — CHANGES_REQUESTED** [review]
You will have to install the following packages `bootstrap` and `react-router-dom` for your code to run.

File: src/index.jsx (hunk 2 above)
Comment (donije09) [reply src/index.jsx:4]: i used command promt

File: src/components/movie-view/movie-view.jsx (hunk 1 above)
Comment (donije09) [reply src/components/movie-view/movie-view.jsx:3]: i used command promt

File: src/components/main-view/main-view.jsx [hunk 3]
```diff
@@ -1,77 +1,159 @@
...
+import { BrowserRouter as Router, Route, Routes, Navigate } from 'react-router-dom';
 import { MovieCard } from '../movie-card/movie-card';
+import { MovieView } from '../movie-view/movie-view';
 import { LoginView } from '../login-view/login-view';
 import { SignupView } from '../signup-view/signup-view';
+import { ProfileView } from '../profile-view/profile-view';
+import { NavigationBar } from '../navigation-bar/navigation-bar';
 
 export const MainView = () => {
   const [movies, setMovies] = useState([]);
```
Comment (truetechcode) [inline src/components/main-view/main-view.jsx:16]: Make the following changes to your `MainView`:
-  This is the correct position.

File: src/components/main-view/main-view.jsx [hunk 4]
```diff
@@ -1,77 +1,159 @@
...
+                    ))}
+                  </Row>
+                ) : (
+                  <p>Loading movies...</p>
+                )
+              ) : (
+                <Navigate to="/login" replace />
+              )
+            } />
+            <Route path="/login" element={<LoginView onLogin={handleLogin} />} />
```
Comment (truetechcode) [inline src/components/main-view/main-view.jsx]: - Make this update to your `login` route.

**Review by truetechcode — CHANGES_REQUESTED** [review]
Make the following changes to your `MainView`:
-  This is the correct position.



- Make this update to your `login` route.

**Review by truetechcode — COMMENTED** [review]
This is good, you should now merge this branch.

**Pull request merged**